boa/src/environment/declarative_environment_record.rs

```diff
@@ -41,9 +42,8 @@ impl EnvironmentRecordTrait for DeclarativeEnvironmentRecord {
         self.env_rec.contains_key(name)
     }
 
-    fn create_mutable_binding(&mut self, name: String, deletion: bool) {
+    fn create_mutable_binding(&mut self, name: String, deletion: bool) -> Result<(), ErrorKind> {
         if self.env_rec.contains_key(&name) {
-            // TODO: change this when error handling comes into play
             panic!("Identifier {} has already been declared", name);
```

Comment:
Should we maintain this panic?

Reply:
Some `panic!`s in the environment module correspond to `assert` in the spec.

Reply:
In that case, I think it's better to use assert:
```rust
assert!(!self.env_rec.contains_key(&name), "identifier {} has already been declared", name);
```

Reply:
What would you suggest for this?

```Rust
if let Some (binding) = self.env_rec.get (name) {
    ...
} else {
   panic! ("Cannot get binding value");
}
```

Reply:
I think that one is good as it is now :)